src/GameLogic/ItemPriceCalculator.cs

```diff
@@ -313,20 +394,62 @@ public long CalculateRepairPrice(Item item, bool npcDiscount)
     }
 
     /// <summary>
-    /// Calculates the buying price of the item, which the player has to pay if he wants to buy the item from a merchant.
+    /// Calculates the final buying price of the item, which the player has to pay if he wants to buy the item from a merchant.
     /// </summary>
     /// <param name="item">The item.</param>
     /// <returns>The buying price.</returns>
-    public long CalculateBuyingPrice(Item item) => CalculateBuyingPrice(item, item.Durability());
+    public long CalculateFinalBuyingPrice(Item item) => RoundPrice(CalculateBuyingPrice(item));
+
+    /// <summary>
+    /// Calculates the final "old" buying price of the item.
+    /// Supposedly in earlier versions jewel reference prices were different, and those were used since for Chaos Weapon and First Wings craftings rate calculations.
+    /// </summary>
+    /// <param name="item">The item.</param>
+    /// <returns>The "old" buying price.</returns>
+    public long CalculateFinalOldBuyingPrice(Item item)
+    {
+        if (SpecialItemOldValueDictionary.TryGetValue(GetId(item.Definition!.Group, item.Definition.Number), out var oldValue))
+        {
+            return RoundPrice(oldValue);
+        }
+        else
+        {
+            return this.CalculateFinalBuyingPrice(item);
+        }
+    }
```

Comment:
The server-side rates now perfectly match the client's. Chaos Weapon and First Wing mixes are a kind of legacy logic.
https://github.com/kyleruss/emu-server/blob/202856a74c905c203b9b2795fd161f564ca8b257/GameServer/Source/ChaosBox.cpp#L207